indico/modules/events/abstracts/forms.py

```diff
@@ -61,3 +63,54 @@ class AbstractSubmissionSettingsForm(IndicoForm):
     authorized_submitters = PrincipalListField(_("Authorized submitters"),
                                                description=_("These users may always submit abstracts, even outside "
                                                              "the regular submission period."))
+
+
+class EmailTemplateRuleFormEdit(IndicoForm):
+    """Form for editing a new e-mail template."""
+
+    title = TextField(_("Title"), [DataRequired()])
+    rules = EmailRuleListField(_("Rules"), [DataRequired()])
+    stop_on_match = BooleanField(_("Stop on match"), [DataRequired()], widget=SwitchWidget(),
+                                 description=_("Do not evaluate any other rules once this one matches."),
+                                 default=True)
+
+    def __init__(self, *args, **kwargs):
+        self.event = kwargs.pop('event')
+        super(EmailTemplateRuleFormEdit, self).__init__(*args, **kwargs)
+        self.rules.event = self.event
+
+    def validate_rules(self, field):
+        dedup_data = {tuple((k, tuple(v)) for k, v in r.viewitems()) for r in field.data}
+        if len(field.data) != len(dedup_data):
+            raise ValidationError(_("There is a duplicate rule"))
+
+
+class EmailTemplateTextFormEdit(IndicoForm):
+    """Form for editing the text of a new e-mail template."""
+
+    reply_to_address = SelectField(_('"Reply to" address'), [DataRequired()])
+    include_submitter = BooleanField(_('Send to submitter'), widget=SwitchWidget())
```

Comment:
unrelated, but I really think we should start adding a `SwitchField` that subclasses `BooleanField` and just sets `widget = SwitchWidget` so we don't have to repeat it all the time (we already have 80 occurrences of `widget=SwitchWidget()`)